docs/site/Troubleshooting.md

```diff
@@ -0,0 +1,115 @@
+binding error is "a binding key not bound", for example:
+
+```
+500 Error: The key 'services.hasher' is not bound to any value in context application
+```
+
+Here are some potential reasons:
+
+1. The key is not bound to any value.
+
+To fix it, you should set the value for the binding key in the application.
+
+For example, go to your application, typically it is in `src/application.ts`
+file:
+
+```ts
+import {Hasher} from '/services/hasher';
+
+export class TodoListApplication extends BootMixin(
+  ServiceMixin(RepositoryMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+    // Add your binding here
+    this.bind(services.hasher).toClass(Hasher);
+  }
+}
+```
+
+2. Incorrect injection key.
+
+When injecting a binding, make sure its name is correct. Especially for the ones
+following LoopBack artifact naming conventions. For example, the automatically
+bound services have their class name as `<ServiceName>Provider` and the binding
+key as `services.<ServiceName>`. So use `@inject('services.UserService')`
+instead of `@inject('services.UserServiceProvider')`.
+
+See more details in
+[Binding documentation page](https://loopback.io/doc/en/lb4/Binding.html).
+
+3. Missing components.
+
+If the binding is exported by a component, ensure the component is installed and
+binding is imported correctly.
+
+4. Compiled files not updated.
+
+See section [compiled files not updated](#compiled-files-not-updated).
+
+### Compiled Files not Updated
+
+The TypeScript application is compiled to JavaScript files and runs. Forgetting
+to rebuild could run into errors. Make sure you rebuild the project before
+re-running the application. By default, if you start the application using
+`npm start`, it rebuilds the project for you.
+
+In some cases, the compiled JS files might not get updated properly even you
+build the project. For example, you deleted a Controller file but still get
+errors about the removed Controller during application startup, it implies the
+compiled JS files probably didn't get updated.
+
+To fix it, run `npm run clean` to delete all the compiled files. By doing so, it
+forces the build to generate the compiled JS files next time when you start the
+application using `npm start` or call `npm run build`.
+
+## Submitting Sample Application for Problem Determination
+
+If you couldn't figure out what is going wrong or would like to report an issue,
+you can submit a simplified application that contains the problematic code using
+the sandbox.
+
+The `/sandbox` directory in the `loopback-next` monorepo can be used to utilize
+the latest source code as symbolically-linked dependencies. See its
+[README](https://github.com/strongloop/loopback-next/tree/master/sandbox) for
+usage instructions.
+
+To submit the sample, you can create a LoopBack application and copy over the
+sandbox files, then publish it to the Github.
```

Comment:
I think this section is partially duplicating the content in https://loopback.io/doc/en/contrib/Reporting-issues.html#loopback-4x-bugs.

Have you considered keeping this section shorter and refer readers to https://loopback.io/doc/en/contrib/Reporting-issues.html#loopback-4x-bugs for more details?

Reply:
@bajtos https://loopback.io/doc/en/contrib/Reporting-issues.html#loopback-4x-bugs doesn't really cover using sandbox to reproduce the bug.